satellite/satellitedb/containment.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"database/sql"
+
+	"storj.io/storj/pkg/audit"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+type containment struct {
+	db *dbx.DB
+}
+
+// Get gets the pending audit by node id
+func (containment *containment) Get(ctx context.Context, id pb.NodeID) (*audit.PendingAudit, error) {
+	if id.IsZero() {
+		return nil, audit.ContainError.New("node ID empty")
+	}
+
+	pending, err := containment.db.Get_PendingAudits_By_NodeId(ctx, dbx.PendingAudits_NodeId(id.Bytes()))
+	if err == sql.ErrNoRows {
+		return nil, audit.ErrContainedNotFound.New(id.String())
+	}
+	if err != nil {
+		return nil, audit.ContainError.Wrap(err)
+	}
+
+	return convertDBPending(pending)
+}
+
+// IncrementPending will create a new pending audit entry, or increase its reverify count if it already exists
+func (containment *containment) IncrementPending(ctx context.Context, pendingAudit *audit.PendingAudit) error {
```

Comment:
What happens if there is already a pending audit for the same node? Do we add a second pending audit, or do we return an error?

Reply:
We don't add a second pending audit, we just increment the existing pending audit's Reverify Count. We don't return an error either. Would it be better if we did?

Reply:
I think we should return an error if there is already a pending audit for a different erasure share. This way we would catch a potential bug in our logic if we are trying the add another pending audit despite of already having one for this node.

Reply:
this makes sense, but I thought the point of having an `IncrementPending` upsert instead of `Create` and `Update` was so that `IncrementPending` could be used for both creating pending audits and then just incrementing their counters if the entry already existed. I thought we decided on this because if one node is not sending audit info for two different requested pieces, then we shouldn't record the second piece's info, we should just increment that node's existing reverify count as penalty.

Reply:
>  if one node is not sending audit info for two different requested pieces

How can this happen in the first place?

If a node fails in audit, we record the audited erasure share as a pending audit in the containment db. If the same node is audited for a different erasure share, we should not audit it for this new erasure share, but for the one we have as a pending audit. If it fails again, we call `IncrementPending`, but we pass the previous pending audit as a param, not a new one.

Reply:
ah you're right, I think I had the chain of events confused. I'll make sure this returns error if there's another pending audit for a different erasure share. do you know if I'll need to add a second query to get that info or if I can add to the existing query?

Reply:
if we want that the caller always verify so, I agree with @kaloyan-raev however when I read this logic and the method comment, I thought that we don't want that the caller had to worry of doing that check, just only call the function.

Reply:
model pending_audits (
	key node_id
^^ NodeID is already unique

Reply:
hm so should I add a check to see if the latest pending audit has the same node ID as another pending audit already in the db, but where the other fields are different? I think that would require calling Get before each IncrementPending. Would that defeat the point of the upsert or is that okay?

Reply:
if `node_id` is already unique, my guess is that the database will return an error if you try to add another row with the same node id.

Reply:
@kaloyan-raev I added a test case for that here and it actually just increments the existing pending audit's reverify count but doesn't error

Reply:
It doesn't return an error because of the `ON CONFLICT` clause however it's better to have a test case for such scenario.

Reply:
I think this happens because we have `ON CONFLICT(node_id) DO UPDATE SET ...` in the SQL statement.

Reply:
I think this happens because we have `ON CONFLICT(node_id) DO UPDATE SET ...` in the SQL statement.

Reply:
okay, I'll go ahead and take that out.